lua/wikis/commons/ThisDay.lua

```diff
@@ -0,0 +1,384 @@
+			ConditionNode(ColumnName('date_day'), Comparator.eq, day),
+			ConditionNode(ColumnName('date'), Comparator.lt, os.date('%Y-%m-%d', os.time() - 86400)),
+			ConditionNode(ColumnName('placement'), Comparator.eq, 1),
+			ConditionNode(ColumnName('opponentname'), Comparator.neq, 'TBD'),
+			ConditionNode(ColumnName('opponentname'), Comparator.neq, 'Definitions'),
+			ConditionTree(BooleanOperator.any)
+				:add{ConditionNode(ColumnName('prizepoolindex'), Comparator.eq, '')}
+				:add{ConditionNode(ColumnName('prizepoolindex'), Comparator.eq, '1')}
+		}
+	conditions:add(Query._multiValueCondition(
+		'liquipediatier',
+		Config.tiers,
+		BooleanOperator.any
+	))
+	conditions:add(Query._multiValueCondition(
+		'liquipediatiertype',
+		Config.tierTypes,
+		BooleanOperator.all
+	))
+
+	local tournamentWinData = mw.ext.LiquipediaDB.lpdb('placement', {
+		limit = 5000,
+		conditions = conditions:toString(),
+		query = 'extradata, pagename, date, icon, icondark, shortname, tournament, series, '
+			.. 'opponentname, opponenttemplate, opponentplayers, opponenttype'
+			.. ', mode, participant, participantflag, participanttemplate', --this line for legacy reasons
+		order = 'date asc, pagename asc'
+	})
+
+	if type(tournamentWinData[1]) == 'table' then
+		return tournamentWinData
+	end
+end
+
+--- build conditions for multi variable
+---@param key string
+---@param values table
+---@param booleanOperator lpdbBooleanOperator
+---@return table?
+function Query._multiValueCondition(key, values, booleanOperator)
+	if Table.isEmpty(values) then
+		return
+	end
+
+	local conditions = ConditionTree(booleanOperator)
+
+	for _, value in pairs(values) do
+		conditions:add{ConditionNode(ColumnName(key), Comparator.eq, value)}
+	end
+
+	return conditions
+end
+
+
+local ThisDay = {}
+
+---@param args table
+---@return Widget
+function ThisDay.run(args)
+	local tournaments = {
+		HtmlWidgets.H3{children = 'Tournaments'},
+		ThisDay.tournament(args)
+	}
+	local birthdays = {
+		HtmlWidgets.H3{children = 'Birthdays'},
+		ThisDay.birthday(args)
+	}
+	local patches = Config.showPatches and {
+		HtmlWidgets.H3{children = 'Patches'},
+		ThisDay.patch(args)
+	} or {}
+	local month, day = ThisDay._readDate(args)
+	local triviaText = Template.safeExpand(
+		mw.getCurrentFrame(),
+		String.interpolate('Liquipedia:This day/${month}/${day}', {month = month, day = day})
+	)
+	local trivia = String.isNotEmpty(triviaText) and {
+		HtmlWidgets.H3{children = 'Trivia'},
+		triviaText
+	} or {}
+	return HtmlWidgets.Fragment{
+		children = WidgetUtil.collect(tournaments, birthdays, patches, trivia)
+	}
+end
+
+--- Get and display birthdays that happened on a given date (falls back to today)
+---@param args {date: string?, month: string|integer|nil, day: string|integer|nil, noTwitter: boolean?}
+---@return string|Widget?
+function ThisDay.birthday(args)
+	local birthdayData = Query.birthday(ThisDay._readDate(args))
+
+	if not birthdayData then
+		return 'There are no birthdays today'
```

Comment:
several wikis suppress this return (plus the header) in `Template:This day`
we should add an option for it

example: https://liquipedia.net/starcraft2/index.php?title=Template:This_day&action=edit